### DISCUSSION TITLE
JSBSim force axis

### DISCUSSION THREAD

**Discussion by matheusdarocha** [JSBSim-Team/jsbsim#730] [General]
According to the JSBSim Reference Manual:

**Structural Frame:** X-axis increases from the nose towards the tail, the Y-axis increases from the fuselage out towards the right (when looking forward from the cockpit), and of course the Z-axis then is positive upwards.

**Body frame** - In JSBSim, the body frame is similar to the structural frame, but rotated 180 degrees about the Y axis, with the origin coincident with the CG.

**Stability frame -** This frame is similar to the body frame, except that the X-axis points into the relative wind vector projected onto the XY plane of symmetry for the aircraft. The Y-axis still points out the right wing, and the Z-axis completes the right-hand system.

**Wind frame -**  This frame is similar to the Stability frame, except that the X-axis points directly into the relative wind. The Z-axis is perpendicular to the X-axis, and remains within the aircraft body axis XZ plane (also called the reference plane). The Y-axis completes a right hand coordinate system.

Running a trim analysis in python using `fdm.do_trim(0)` and printing the results as below:

```python
print("body",fdm['forces/fbx-aero-lbs'],fdm['forces/fby-aero-lbs'],fdm['forces/fbz-aero-lbs'])
print("wind",fdm['forces/fwx-aero-lbs'],fdm['forces/fwy-aero-lbs'],fdm['forces/fwz-aero-lbs'])
print("Stab",fdm['forces/fsx-aero-lbs'],fdm['forces/fsy-aero-lbs'],fdm['forces/fsz-aero-lbs'])
```
we got:

```sh
body -11436.386120435793 0.0 -48342.822873241166
wind 11349.303086115828 0.0 48363.34118222962
Stab 11349.303086115828 0.0 48363.34118222962
```

According to the manual, the Z axis from body wind and stab point "from the aircraft to the earth".
Why does the body axis is giving a different signal than the stability and wind?

**Comment by bcoconni** [comment]
At first, I'd say it looks like you're flying rearward 😄
Would you mind providing the velocities in the body frame ?

**Reply by matheusdarocha** [reply]
Sure! the velocites are:

```sh
velocities/u-fps = 459.589555594661
velocities/v-fps = 0
velocities/w-fps = -0.8277132256758307
```

**Comment by seanmcleod** [comment]
> Why does the body axis is giving a different signal than the stability and wind?

So, in terms of the stability and wind axes the lift and drag forces are defined as positive for "up" and "rearward".

See - https://jsbsim-team.github.io/jsbsim-reference-manual/mypages/user-manual-frames-of-reference/

![image](https://user-images.githubusercontent.com/824856/191499632-d5a472fd-dfe0-4872-a354-0e7534c30e33.png)

So `forces/fsx-aero-lbs` is outputting the drag force and `forces/fwz-aero-lbs` is outputting the lift force and hence the difference in sign between the body axis compared to the stability and wind axes.

**Reply by matheusdarocha** [reply]
Thanks Sean!

**Reply by bcoconni** [reply]
Hmmm... In the picture above the lift $L$ is pointing in the direction opposite to both axes $z_A$ et $z_B$ so it should have the same sign in the 2 frames, shouldn't it ?

**Comment by seanmcleod** [comment]
The impression I have is that it is simply convention that when talking about forces in the stability and wind axes the resultant lift and drag forces are defined as positive opposite to the sign of the relevant axes, but when referring to the same forces in the body axes the sign of the forces match sign of the individual axes.

**Comment by seanmcleod** [comment]
For example from a textbook section "Aerodynamic Derivatives in Body Fixed Axes":

![image](https://user-images.githubusercontent.com/824856/191589405-07089006-2965-4ac4-9dff-9bccb0168258.png)
